app/models/commitment.rb

```diff
@@ -34,7 +34,9 @@ class Commitment < ApplicationRecord
   belongs_to :user, optional: true
 
   accepts_nested_attributes_for :links, reject_if: ->(attributes) { attributes['url'].blank? }, allow_destroy: true
-  accepts_nested_attributes_for :progress_documents, reject_if: ->(attributes) { attributes['document'].blank? }, allow_destroy: true
+  accepts_nested_attributes_for :progress_documents, reject_if: ->(attributes) {
```

Comment:
i don't think these reject_if work (maybe reject_if doesn't work at all for has_many?), and now you have the front end submitting the params correctly we don't need them, so can be replaced with: ```  accepts_nested_attributes_for :links, allow_destroy: true
  accepts_nested_attributes_for :progress_documents, allow_destroy: true```